### PR TITLE
fix(build): install luajit modules on windows

### LINKED ISSUE #16659: [LuaJIT] bundled installation is missing 'jit*.' modules on windows
### Neovim version (nvim -v)

v0.7.0-dev+stable-68-ge9a0bd532

### Vim (not Nvim) behaves the same?

NA

### Operating system/version

Windows 11

### Terminal name/version

NA

### $TERM environment variable

NA

### Installation

from source

### How to reproduce the issue

build from source, for more info check the discussion in https://github.com/neovim/neovim/pull/16631

### Expected behavior

This call shouldn't produce any erors

```console
$ .deps\usr\bin\luajit -e "require('jit.bcsave')"
```

### Actual behavior

```console
.. (command line):1: module 'jit.bcsave' not found: ..
```

### PR DESCRIPTION
**Pull request by kylo252**
Fixes #16659

### CODE REVIEW

**Review by jamessan — APPROVED** [review]

File: third-party/cmake/BuildLuajit.cmake [hunk 1]
```diff
@@ -138,8 +140,10 @@ elseif(MSVC)
       # Luv searches for luajit.lib
       COMMAND ${CMAKE_COMMAND} -E copy ${DEPS_BUILD_DIR}/src/luajit/src/lua51.lib ${DEPS_INSTALL_DIR}/lib/luajit.lib
       COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/include/luajit-2.1
-      COMMAND ${CMAKE_COMMAND} -DFROM_GLOB=${DEPS_BUILD_DIR}/src/luajit/src/*.h -DTO=${DEPS_INSTALL_DIR}/include/luajit-2.1 -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CopyFilesGlob.cmake)
-
+      COMMAND ${CMAKE_COMMAND} -DFROM_GLOB=${DEPS_BUILD_DIR}/src/luajit/src/*.h -DTO=${DEPS_INSTALL_DIR}/include/luajit-2.1 -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CopyFilesGlob.cmake
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin/lua/jit
+      COMMAND ${CMAKE_COMMAND} -E copy_directory ${DEPS_BUILD_DIR}/src/luajit/src/jit ${DEPS_INSTALL_DIR}/bin/lua/jit
+            )
```
Comment (erw7) [inline third-party/cmake/BuildLuajit.cmake:146]:
```suggestion
      )
```

File: third-party/cmake/BuildLuajit.cmake [hunk 2]
```diff
@@ -124,6 +124,8 @@ elseif(MINGW)
 	    COMMAND ${CMAKE_COMMAND} -E copy ${DEPS_BUILD_DIR}/src/luajit/src/libluajit.a ${DEPS_INSTALL_DIR}/lib
 	    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/include/luajit-2.1
 	    COMMAND ${CMAKE_COMMAND} -DFROM_GLOB=${DEPS_BUILD_DIR}/src/luajit/src/*.h -DTO=${DEPS_INSTALL_DIR}/include/luajit-2.1 -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CopyFilesGlob.cmake
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin/lua/jit
+      COMMAND ${CMAKE_COMMAND} -E copy_directory ${DEPS_BUILD_DIR}/src/luajit/src/jit ${DEPS_INSTALL_DIR}/bin/lua/jit
           )
```
Comment (erw7) [inline third-party/cmake/BuildLuajit.cmake:129]:
```suggestion
	    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin/lua/jit
	    COMMAND ${CMAKE_COMMAND} -E copy_directory ${DEPS_BUILD_DIR}/src/luajit/src/jit ${DEPS_INSTALL_DIR}/bin/lua/jit
	    )
```

File: third-party/cmake/BuildLuajit.cmake [hunk 3]
```diff
@@ -124,6 +124,8 @@ elseif(MINGW)
 	    COMMAND ${CMAKE_COMMAND} -E copy ${DEPS_BUILD_DIR}/src/luajit/src/libluajit.a ${DEPS_INSTALL_DIR}/lib
 	    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/include/luajit-2.1
 	    COMMAND ${CMAKE_COMMAND} -DFROM_GLOB=${DEPS_BUILD_DIR}/src/luajit/src/*.h -DTO=${DEPS_INSTALL_DIR}/include/luajit-2.1 -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/CopyFilesGlob.cmake
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin/lua/jit
+      COMMAND ${CMAKE_COMMAND} -E copy_directory ${DEPS_BUILD_DIR}/src/luajit/src/jit ${DEPS_INSTALL_DIR}/bin/lua/jit
```
Comment (zeertzjq) [inline third-party/cmake/BuildLuajit.cmake:128]: Maybe these lines' indent should be changed as well.

**Comment by erw7** [comment]
Do you need to make this a standalone PR? I think we can accommodate with #16631 or follow up PR. Until #16631 is merged, the `jit` module should not be needed to build neovim.

**Comment by kylo252** [comment]
> Do you need to make this a standalone PR? I think we can accommodate with #16631 or follow up PR. Until #16631 is merged, the `jit` module should not be needed to build neovim.

I got the general impression that the Windows part was going to be shelved, so I figured it's better to get the fix in at least since it's independent enough, and I'm pretty sure I came across this issue at some point or another.

**Comment by gpanders** [comment]
This seems to affect platforms other than Windows as well, see #16709.

**Pull request merged**